src/contrib/cluster/Akka.Cluster.Tools.Tests/Singleton/MemberAgeOrderingSpec.cs

```diff
@@ -0,0 +1,89 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="MemberAgeOrderingSpec.cs" company="Akka.NET Project">
+//      Copyright (C) 2009-2022 Lightbend Inc. <http://www.lightbend.com>
+//      Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Akka.Actor;
+using Akka.Cluster.Tools.Singleton;
+using Akka.Util;
+using FluentAssertions;
+using Xunit;
+
+namespace Akka.Cluster.Tools.Tests.Singleton
+{
+    public class MemberAgeOrderingSpec
+    {
+        [Fact(DisplayName = "MemberAgeOrdering should sort based on UpNumber")]
+        public void SortByUpNumberTest()
+        {
+            var members = new SortedSet<Member>(MemberAgeOrdering.DescendingWithAppVersion)
+            {
+                Create(Address.Parse("akka://sys@darkstar:1112"), upNumber: 3),
+                Create(Address.Parse("akka://sys@darkstar:1113"), upNumber: 1),
+                Create(Address.Parse("akka://sys@darkstar:1111"), upNumber: 9),
+            };
+
+            var seq = members.ToList();
+            seq.Count.Should().Be(3);
+            seq[0].Should().Be(Create(Address.Parse("akka://sys@darkstar:1113"), upNumber: 1));
+            seq[1].Should().Be(Create(Address.Parse("akka://sys@darkstar:1112"), upNumber: 3));
+            seq[2].Should().Be(Create(Address.Parse("akka://sys@darkstar:1111"), upNumber: 9));
+        }
+        
+        [Fact(DisplayName = "MemberAgeOrdering should sort based on Address if UpNumber is the same")]
```

Comment:
Not possible for UpNumber to be the same - that'd be the result of data corruption.

Reply:
Its part of the `MemberAgeOrdering` code, it defaults to member `Address` ordering if the `UpNumber` is the same.
Since this is a spec for the comparator, I might as well make sure that it works.

Reply:
ok, that's fine.